drivers/npmx/npmx_shell.c

```diff
@@ -809,6 +809,85 @@ static int cmd_ntc_resistance_hot_get(const struct shell *shell, size_t argc, ch
 	return cmd_ntc_resistance_get(shell, argc, argv, npmx_charger_hot_resistance_get);
 }
 
+static int cmd_charger_discharging_current_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	uint16_t current;
+	npmx_error_t err_code = npmx_charger_discharging_current_get(charger_instance, &current);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Value: %u mA.", current);
+	} else {
+		shell_error(shell, "Error: unable to read trickle voltage value.");
+	}
+
+	return 0;
+}
+
+static int cmd_charger_discharging_current_set(const struct shell *shell, size_t argc, char **argv)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing current value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint16_t discharging_current = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT16_MAX);
+
+	if (err != 0) {
+		shell_error(shell, "Error: current has to be an integer.");
+		return 0;
+	}
+
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	uint32_t modules_mask;
+	npmx_error_t err_code = npmx_charger_module_get(charger_instance, &modules_mask);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to get charger module status.");
+		return 0;
+	}
+
+	if ((modules_mask & NPMX_CHARGER_MODULE_CHARGER_MASK) != 0) {
+		shell_error(shell, "Error: charger must be disabled to set discharging current.");
+		return 0;
+	}
+
+	err_code = npmx_charger_discharging_current_set(charger_instance, discharging_current);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to set discharging current value.");
+	}
+
+	err_code = npmx_charger_discharging_current_get(charger_instance, &discharging_current);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %d mA.", discharging_current);
+		shell_print(shell, "Set value may be different than requested.");
```

Comment:
Is this due to `Maximum discharging current in milliamperes in a range from 270 to 1340 with a step of 2.`?
If so, the value should be first checked to make sure it fits the requirements.

Reply:
There is a calculation: 
`Code = round (Current (mA)/3.23).`

The value is checked in the driver.

Reply:
I think we should add checking if the exact value can be set. Probably it would be the best to do it in the driver.
Let's discuss offline tomorrow.